### ISSUE TITLE
Footnotes: generates a section without heading causing validation errors

### ISSUE THREAD

**Issue by danielfdickinson** [gohugoio/hugo#9756]
### What version of Hugo are you using (`hugo version`)?

`hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended linux/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=gohugoio`

### Does this issue reproduce with the latest release?

yes

## Preface

I initially submitted this some time ago to https://github.com/yuin/goldmark/issues/256 not being aware of the policy around submitting to Hugo first for Hugo usage of `goldmark`, so this is correcting the process.

## Details

https://github.com/yuin/goldmark/blob/829d874034a5048e1a5c6354dcd8b40e1e41c3a2/extension/footnote.go#L579

causes

| 175 | \<section class="footnotes" role="doc-endnotes"> \<hr> | Section lacks heading. Consider using “h2”-“h6” elements to add identifying headings to all sections. |

in W3C http://validator.w3.org/unicorn

There does not appear to be a way to configure the tag to use (e.g.) 'div' when using HTML5. I'm probably missing something, but I don't see a way to override this without changing ``footnote.go`` source file.

I do note that in the referenced https://michelf.ca/projects/php-markdown/extra/#footnotes that the example output given is

```html
<p>That's some text with a footnote.
   <sup id="fnref:1"><a href="#fn:1" class="footnote-ref" role="doc-noteref">1</a></sup></p>

<div class="footnotes" role="doc-endnotes">
<hr />
<ol>

<li id="fn:1" role="doc-endnote">
<p>And that's the footnote.
   <a href="#fnref:1" class="footnote-backref" role="doc-backlink">&#8617;</a></p>
</li>

</ol>
</div>
``` 

which uses a div (but also looks like the XHTML version).

### Additional information

For what it's worth here are some articles on the status of outlines (sections) and headings:

<https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#multiple_h1_elements_on_one_page> points to

[Computer says no to the HTML5 outline](https://html5doctor.com/computer-says-no-to-html5-document-outline/)
[A decade + a year of heading backwards](https://html5accessibility.com/stuff/2021/03/21/a-decade-a-year-of-heading-backwards/)

Basically, it's a mess of spec that was never implemented and docs never updated to reflect implementation.
